app/lib/actions/katello/event_queue/monitor.rb

```diff
@@ -0,0 +1,60 @@
+module Actions
+  module Katello
+    module EventQueue
+      class Monitor < Actions::EntryAction
```

Comment:
The default rescue_strategy is to pause the action. Iirc paused actions still hold the singleton lock. Using any of the other two strategies which would cause the lock to be unlocked on failure would probably be better idea for more hands-off operation.